translate file finders.zh-cn

skyfackr opened the pull request:
**Thanks for contributing to the Selenium site and documentation!**
**A PR well described will help maintainers to review and merge it quickly**

Before submitting your PR, please check our [contributing](https://selenium.dev/documentation/en/contributing/) guidelines.
Avoid large PRs, and help reviewers by making them as simple and short as possible.

translate file finders.zh-cn
<!--- Provide a general summary of your changes in the Title above -->

### Description
<!--- Describe your changes in detail -->
translate the file /webdriver/elements/finders.zh-cn.md to Chinese

Effective at /zh-cn/documentation/webdriver/elements/finders/

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
The original finder.zh-cn.md was still written in English. This PR translates it.

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [ ] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [x] Added new translation (and I also added a notice to each document missing translation)

### Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [x] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.
<!--- Provide a general summary of your changes in the Title above -->

skyfackr committed:
translate file finders.zh-cn

translate the file /webdriver/elements/finders.zh-cn.md to Chinese

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/SeleniumHQ/seleniumhq.github.io?pullRequest=1432) <br/>All committers have signed the CLA.

alaahong commented on website_and_docs/content/documentation/webdriver/elements/finders.zh-cn.md:
```diff
@@ -30,17 +29,15 @@ let's consider this HTML snippet:
+### 在整个DOM中查找元素
+
+当在驱动实例上调用`find element`方法时，它会返回DOM中与所提供的定位器匹配的第一个元素的引用。这个引用可以被保存用于之后对此元素的操作。
+在上面的HTML示例中，有两个元素的类名都是 "tomatoes"，他们分别在两个列表中。以下代码将返回第一个列表，即 "vegetables "列表中的"tomatos"元素引用。
```
Seems it will fetch below element, please revise your additional word
![image](https://github.com/SeleniumHQ/seleniumhq.github.io/assets/3264250/de93856b-d7b8-4c44-a972-b895f20d537f)

skyfackr replied:
Hello,

Sorry but I cannot understand what you mean. This paragraph fetches the element in this picture, but I don't know what you mean for additional word?

alaahong replied:
Hi , you just highlight one element which is not from original content or element source.

skyfackr replied:
hello,

Do you mean "tomato" is not from the original content? However, I just checked them, and I think there is no additional content in my transaction. I just highlighted "the return element is from the first list." Do you think it is spare?

alaahong commented on website_and_docs/content/documentation/webdriver/elements/finders.zh-cn.md:
```diff
@@ -64,14 +61,12 @@ val vegetable: WebElement = driver.findElement(By.className("tomatoes"))
 
 
-### Evaluating a subset of the DOM
+### 在部分DOM中查找元素
```
how about “子集” as this under the DOM tree ?

skyfackr replied:
Hello,

You are right. I will add it

alaahong commented on website_and_docs/content/documentation/webdriver/elements/finders.zh-cn.md:
```diff
@@ -64,14 +61,12 @@ val vegetable: WebElement = driver.findElement(By.className("tomatoes"))
-### Evaluating a subset of the DOM
+### 在部分DOM中查找元素
+
+以上代码无法获取在样例第二个列表中的"tomato"元素。如果知道目标元素所属位置且希望获取指定范围里的元素而不是整个上下文中的第一个，可以进行部分搜索。
```
Did you miss content here?

skyfackr replied:
Hello,

If you mean `Rather than finding a unique locator in the entire DOM, it is often useful to narrow the search to the scope of another located element. ` 

I translated it as the sentence on line 66. The original sentence wants to say that "search in subset DOM" is better. I use the sentence `如果知道目标元素所属位置且希望获取指定范围里的元素而不是整个上下文中的第一个，可以进行部分搜索。` , which is more clear for Chinese audiences reading.

If you think it is better leaving a direct transaction, please let me know.

Thanks

alaahong replied:
This is the same issue as tomatoes...

alaahong commented on website_and_docs/content/documentation/webdriver/elements/finders.zh-cn.md:
```diff
@@ -101,23 +96,19 @@ val fruit = fruits.findElement(By.className("tomatoes"))
+为了略微提高性能，我们可以使用 CSS 或 XPath 在单个命令中查找该元素。请查看在[最佳实践]({{< ref "/documentation/test_practices/encouraged" >}})中的[使用定位器的提示]({{< ref "/documentation/test_practices/encouraged/locators" >}})部分。
 
-For this example, we'll use a CSS Selector:
+如下所示，我们通过CSS路径定位器查找元素：
```
For CSS, it's usually to be called "Selector" not the "Locator" as original context

skyfackr replied:
Hello,

I will change the transaction as "selector"

alaahong commented on website_and_docs/content/documentation/webdriver/elements/finders.zh-cn.md:
```diff
@@ -279,10 +268,9 @@ fun main() {
 
-It is used to find the list of matching child WebElements within the context of parent element.
-To achieve this, the parent WebElement is chained with 'findElements' to access child elements
+类`WebElement`本身支持`findElements`函数。当需要查找一个父节点之下的某个元素，且拥有父节点元素引用时，可以通过对父节点执行查找获取目标元素（或元素列表）。
```
Seems missed some content here?

skyfackr replied:
Hello,

I didn't find any missed content here. Could you please give me some tips?

Thanks!

alaahong replied:
Well, finally get your context after reviewed  several times... 
You can try to rewrite the English version as you bring many self parts...

alaahong replied:
Original English version is rather clear,  "A is XXX and you can work with A on XXX... "

skyfackr replied:
Hello,

I removed all self parts

alaahong reviewed (changes requested):
Thanks for your commit, we're happy to see you as the first-time contributor.
Can you help to update as the comment? 

And meanwhile ,as community guide, it's better to add line break [[reference](https://www.selenium.dev/documentation/about/contributing/#step-4-commit)]
```
The first line should be around 50 characters or less and contain a short description of the change.
Keep the second line blank.
Wrap all other lines at 72 columns.
Include Fixes #N, where N is the issue number the commit fixes, if any.
```

skyfackr commented:
> 

Hello,

I didn't find any existing issue relating to this document translation. And I think this is not complex work, so I didn't create a new issue and just submitted this PR without noting the issue number in the PR comment. If there does have one issue or you think I better create one first, please let me know. Thanks!

alaahong commented:
> > 
> 
> Hello,
> 
> I didn't find any existing issue relating to this document translation. And I think this is not complex work, so I didn't create a new issue and just submitted this PR without noting the issue number in the PR comment. If there does have one issue or you think I better create one first, please let me know. Thanks!

Hi，no need create issue for such translation, it's better to  follow "Wrap all other lines at 72 columns." which means do not put all into one line.

 For your translated parts, please check the comment on previous commit.

skyfackr committed:
Update finders.zh-cn as comment

update as comment #2 and #4

add `DOM子集` as transaction of `subset of DOM`

change `CSS定位器` to `CSS选择器`

skyfackr committed:
Update finders.zh-cn as comment #5

update as comment #5
#5 -> change to transaction without proofreading

alaahong committed:
Merge branch 'trunk' into translate

harsha509 commented:
Hi @skyfackr ,

Did you get a chance to look into the above comments?

skyfackr commented:
Hello sir,

Sorry for late reply. I am unsure which comment you mean? If you mean those
comments from Mr. ian I have read

Sri Harsha ***@***.***> 于2023年10月7日周六 14:35写道：

> Hi @skyfackr <https://github.com/skyfackr> ,
>
> Did you get a chance to look into the above comments?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1432#issuecomment-1751790623>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKQSKAGIHR5EK7L5WFVF54LX6GOHZAVCNFSM6AAAAAA2IQWFNOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONJRG44TANRSGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

alaahong committed:
Merge branch 'trunk' into translate

alaahong commented:
> Hello sir, Sorry for late reply. I am unsure which comment you mean? If you mean those comments from Mr. ian I have read Sri Harsha ***@***.***> 于2023年10月7日周六 14:35写道：
> […](#)
> Hi @skyfackr <https://github.com/skyfackr> , Did you get a chance to look into the above comments? — Reply to this email directly, view it on GitHub <[#1432 (comment)](https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1432#issuecomment-1751790623)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AKQSKAGIHR5EK7L5WFVF54LX6GOHZAVCNFSM6AAAAAA2IQWFNOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONJRG44TANRSGM> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Hi skyfackr,

Are you still there?
Please read below part if any confused on previous comment.  
https://www.selenium.dev/documentation/about/contributing/#step-4-commit

As this commit is long long ago, can you try to pull latest code if still willing to translate this?

skyfackr commented:
Hello,

It is a long time ago but i am still alive. I can see it later today or
this week.
ian zhang ***@***.***>于2025年3月24日 周一09:09写道：

> Hello sir, Sorry for late reply. I am unsure which comment you mean? If
> you mean those comments from Mr. ian I have read Sri Harsha *@*.
>
> *> 于2023年10月7日周六 14:35写道： … <#m_-4089429375914988812_> Hi @skyfackr
> <https://github.com/skyfackr> https://github.com/skyfackr
> <https://github.com/skyfackr> , Did you get a chance to look into the above
> comments? — Reply to this email directly, view it on GitHub <#1432
> (comment)
> <https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1432#issuecomment-1751790623>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AKQSKAGIHR5EK7L5WFVF54LX6GOHZAVCNFSM6AAAAAA2IQWFNOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONJRG44TANRSGM
> <https://github.com/notifications/unsubscribe-auth/AKQSKAGIHR5EK7L5WFVF54LX6GOHZAVCNFSM6AAAAAA2IQWFNOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONJRG44TANRSGM>
> . You are receiving this because you were mentioned.Message ID: @.*>
>
> Hi skyfackr,
>
> Are you still there?
> Please read below part if any confused on previous comment.
> https://www.selenium.dev/documentation/about/contributing/#step-4-commit
>
> As this commit is long long ago, can you try to pull latest code if still
> willing to translate this?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1432#issuecomment-2748081013>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKQSKAC7R25U57YHAQWJ6I32V77ZLAVCNFSM6AAAAABZU3YEACVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBYGA4DCMBRGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: alaahong]*alaahong* left a comment
> (SeleniumHQ/seleniumhq.github.io#1432)
> <https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1432#issuecomment-2748081013>
>
> Hello sir, Sorry for late reply. I am unsure which comment you mean? If
> you mean those comments from Mr. ian I have read Sri Harsha *@*.
>
> *> 于2023年10月7日周六 14:35写道： … <#m_-4089429375914988812_> Hi @skyfackr
> <https://github.com/skyfackr> https://github.com/skyfackr
> <https://github.com/skyfackr> , Did you get a chance to look into the above
> comments? — Reply to this email directly, view it on GitHub <#1432
> (comment)
> <https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1432#issuecomment-1751790623>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AKQSKAGIHR5EK7L5WFVF54LX6GOHZAVCNFSM6AAAAAA2IQWFNOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONJRG44TANRSGM
> <https://github.com/notifications/unsubscribe-auth/AKQSKAGIHR5EK7L5WFVF54LX6GOHZAVCNFSM6AAAAAA2IQWFNOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONJRG44TANRSGM>
> . You are receiving this because you were mentioned.Message ID: @.*>
>
> Hi skyfackr,
>
> Are you still there?
> Please read below part if any confused on previous comment.
> https://www.selenium.dev/documentation/about/contributing/#step-4-commit
>
> As this commit is long long ago, can you try to pull latest code if still
> willing to translate this?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1432#issuecomment-2748081013>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKQSKAC7R25U57YHAQWJ6I32V77ZLAVCNFSM6AAAAABZU3YEACVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBYGA4DCMBRGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

diemol commented:
This went stale, closing this.
